models/dvmdostem/R/model2netcdf.dvmdostem.R

```diff
@@ -227,81 +229,93 @@ model2netcdf.dvmdostem <- function(outdir, runstart, runend, pecan_requested_var
+    lond <- ncdf4::ncdim_def(
+      name = "lon",
+      units = "degrees_east",
+      vals = c(1), # <=== read from dvmdostem file! see dvmdostem issue #342
+      longname = "coordinate_longitude"
+    )
+
+    latd <- ncdf4::ncdim_def(
+      name = "lat",
+      units = "degrees_north",
+      vals = c(1), # <=== read from dvmdostem file! see dvmdostem issue #342
+      longname = "coordinate_latitude"
+    )
 
     if (length(monthly_dvmdostem_outputs) > 0) {
       # last day of each month
       timed_vals <- cumsum(sapply(seq(12), function(x) lubridate::days_in_month(x)))
     } else {
       timed_vals <- c(0)
     }
-    timed <- ncdf4::ncdim_def(name='time',
-                              units=paste0("days since ", all_yrs[i]),
-                              vals=timed_vals,
-                              unlim=TRUE,
-                              longname="time",
-                              calendar='365_day')
+    timed <- ncdf4::ncdim_def(
+      name = "time",
+      units = paste0("days since ", all_yrs[i]),
+      vals = timed_vals,
+      unlim = TRUE,
+      longname = "time",
+      calendar = "365_day"
+    )
 
-    out_nc_dims <- list(lon=lond, lat=latd, time=timed) # dimension order: X, Y, time
+    out_nc_dims <- list(lon = lond, lat = latd, time = timed) # dimension order: X, Y, time
 
     PEcAn.logger::logger.info("Creating variables for new PEcAn style files...")
 
     newvars <- c() # Not very efficient, would be better to pre-allocate space
     j <- 0
-    for (name in pecan_requested_vars){
+    for (name in pecan_requested_vars) {
       j <- j + 1
       print(paste0("Creating variable named: ", name))
-      ncvar <- ncdf4::ncvar_def(name = name,
-                                units = vmap_reverse[[name]][["newunits"]],
-                                longname = vmap_reverse[[name]][["longname"]],
-                                dim = out_nc_dims, -999, prec = "double")
+      ncvar <- ncdf4::ncvar_def(
+        name = name,
+        units = vmap_reverse[[name]][["newunits"]],
+        longname = vmap_reverse[[name]][["longname"]],
+        dim = out_nc_dims, -999, prec = "double"
+      )
       newvars[[j]] <- ncvar
     }
 
     ncout <- ncdf4::nc_create(file.path(outdir, paste0(as.character(lubridate::year(all_yrs[i])), ".nc")), newvars)
     # extract variable and long names to VAR file for PEcAn visibility
     # THIS NEEDS TO BE KEPT AND USED FOR PROPER PLOTTING
-    write.table(sapply(ncout$var, function(x) { x$longname }),
-                file = file.path(outdir,paste0(as.character(lubridate::year(all_yrs[i])), ".nc.var")),
-                col.names = FALSE,
-                row.names = TRUE,
-                quote = FALSE)
+    write.table(sapply(ncout$var, function(x) {
+      x$longname
+    }),
+    file = file.path(outdir, paste0(as.character(lubridate::year(all_yrs[i])), ".nc.var")),
+    col.names = FALSE,
+    row.names = TRUE,
+    quote = FALSE
+    )
```

Comment:
Here's a place I don't like the styler bot's output, but this looks like a corner case. Extending the test: I'll propose a change, accept it, then restyle and see what the bot does. My prediction: It reverts my change completely. My hope: It keeps it, possibly with minor whitespace adjustments.

```suggestion
    write.table(
      sapply(ncout$var, function(x) {x$longname}),
      file = file.path(outdir, paste0(as.character(lubridate::year(all_yrs[i])), ".nc.var")),
      col.names = FALSE,
      row.names = TRUE,
      quote = FALSE)
```

Reply:
/style